v2-apache/conf/extra/httpd-vhosts.conf

```diff
@@ -29,6 +29,18 @@ SSLProxyCACertificateFile ${PROXY_SSL_CA_CERT}
 
 UseCanonicalName on
 
+<VirtualHost *:${PORT}>
+    RewriteEngine On
+    RewriteCond %{REQUEST_URI} !^/\.well\-known/acme\-challenge/
+    RewriteCond %{ENV:APACHE_ALWAYS_TLS_REDIRECT} on
+    RewriteCond %{HTTP_HOST} ^(.*):${PORT}$ [NC]
+    RewriteRule ^(.*)$ https://%1:${SSL_PORT}$1 [L,R=301]
+
+    RewriteCond %{REQUEST_URI} !^/\.well\-known/acme\-challenge/
+    RewriteCond %{ENV:APACHE_ALWAYS_TLS_REDIRECT} on
+    RewriteRule ^(.*)$ https://%{HTTP_HOST}:${SSL_PORT}$1 [L,R=301]
```

Comment:
Why do you have two different rewrites here? Both end up redirecting to the same `(host, port)` tuple, correct?

Reply:
Yes. But it depends a lot if the PORT where the server is running is the default or a different one. That's why you need both rewrites.

Reply:
Ok. I don't understand it but I trust you know why :)